### ISSUE TITLE
Bug with filter_slider and dates

### ISSUE THREAD

**Issue by bhogan-mitre** [rstudio/crosstalk#65]
I encountered an obscure error with `filter_slider` applied to an object of class `Date`.

Here is a minimal example to reproduce it:
https://gist.github.com/bhogan-mitre/81d1eaffc56deb342e8dbff7f2d3194f

If the `filter_slider` of `Date` comes second in the list of inputs, it throws an error:
```
Uncaught TypeError: Cannot read property 'utc' of undefined
```

If the inputs are reversed so the `filter_slider` is first, it renders fine.
```
filter_slider("date", "Date", shared_mtcars, ~date, width = "100%")
filter_checkbox("cyl", "Cylinders", shared_mtcars, ~cyl, inline = TRUE)
```

Note that I am using this with `flexdashboard`, so that may be a contributing factor as well.

**Comment by enricodata** [comment]
I am having the same issue, did you find any solutions?

**Comment by bhogan-mitre** [comment]
No, I have not found any solution other than the workaround described above to reorder the inputs.

**Comment by papapalpatine** [comment]
Any other insights on how to minimize filter_slider issues with dates? I've been running into a lot of trouble with them -- see [here](https://github.com/rstudio/crosstalk/issues/69).

**Comment by cjagoe110** [comment]
Think I've come across a different, but related error.  I have a filter_slider using dates.  When I run the R chunks, it renders and functions correctly.  However, once I knit to HTML the slider becomes an empty box.  I have tried reordering my filters, only keeping the slider, moving into and out of the same R chunk, but all have given me the same results.
![image](https://user-images.githubusercontent.com/73593070/98849116-b5a56180-2420-11eb-91b6-41d5cddcac2b.png)
![image](https://user-images.githubusercontent.com/73593070/98849159-c6ee6e00-2420-11eb-9bc4-a9b6322739f8.png)

**Comment by rgarcia-cat** [comment]
I have encountered the same error and have discovered that it is an html dependency issue. 

The strftime js library is called after the crosstalk library. Changing the order of the html <script> tags works correctly.
```
(before) <script src="site_libs/strftime-0.9.2/strftime-min.js"></script>
(after) <script src="site_libs/crosstalk-1.0.1/js/crosstalk.min.js"></script>
```
In the documentation it is like this, but when rendering, the order is modified.

In my case, I have created an R script that I run after rendering the html.
I have not found another type of solution, since whenever I added elements in the head before rendering, when rendering it was modified.
It is not the best solution but it works...

R script code that fixes the error:
```
library(xml2)

rawHTML = paste(readLines("./docs/index.html"), collapse = "\n")
index = read_html(rawHTML)

rawHTML = '<script src="site_libs/strftime-0.9.2/strftime-min.js"></script>'
custom = read_xml(rawHTML)

xml_add_sibling(xml_children(xml_children(xml_root(xml_root(index)))[[1]])[[1]], xml_root(custom), .where = "before")

write_html(index, "./docs/index.html", optionc="format")
```

**Comment by cpsievert** [comment]
A more minimal reprex:

````
---
title: "Reprex"
output: 
  html_document:
    self_contained: false
---


```{r cars}
library(crosstalk)
data("economics", package = "ggplot2")

mtcars$cyl <- as.factor(mtcars$cyl)
htmltools::tagList(
  filter_select("mag", "Select cylinder", SharedData$new(mtcars), ~cyl),
  filter_slider("date", "Select date", SharedData$new(economics), ~date)
)
```

````

**Comment by asadow** [comment]
Also had this issue.

**Comment by jonathanmburns** [comment]
I had this issue as well with html.

**Comment by roaldarbol** [comment]
Using the solution from https://github.com/rstudio/crosstalk/issues/65#issuecomment-806767100 made it work for me in a Quarto document. 

```yml
---
title: My dashboard
format: 
  html:
    header-includes: |
      <script src="site_libs/strftime-0.9.2/strftime-min.js"></script>
      <script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
---
```
